File: Graph Algorithms in Genome Sequencing/Genome seq/Generate Contigs from a Collection of Reads/contigs.py
#python3
import sys


def contigs(patterns):
    """Finds all contigs in a de Bruijn graph made from patterns

    Args:
        patterns:       a list of strings containing all k-mers to
                        be used to construct a de Bruijn graph

    Returns:
        a string with all the contigs in the format specified by
        the problem description
    """
    # TODO: your code here
    return ""


if __name__ == "__main__":
    patterns = [line.strip() for line in sys.stdin if line.strip()]

    print(contigs(patterns))

File: Graph Algorithms in Genome Sequencing/Genome seq/Generate Contigs from a Collection of Reads/Contigs.java
import java.util.Scanner;
import java.util.List;
import java.util.ArrayList;

public class Contigs {
    /*
     * Finds all contigs in a de Bruijn graph made from patterns
     *
     * @param patterns      a List of Strings containing all k-mers to
     *                      be used to construct the de Bruijn graph for
     *                      this problem
     *
     * @return a String with all the contigs in the format specified in the
     * problem description
     */
    public static String contigs(List<String> patterns) {
        // TODO: your code here
        return "";
    }

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);

        List<String> patterns = new ArrayList<>();
        while (stdin.hasNextLine()) patterns.add(stdin.nextLine());

        System.out.println(contigs(patterns));
    }
}
